src/List.mo

```diff
@@ -625,8 +756,29 @@ module {
   ///
   /// *Runtime and space assumes that `equal` runs in `O(1)` time and space.
   public func indexOf<T>(list : List<T>, equal : (T, T) -> Bool, element : T) : ?Nat {
-    // inlining would save 10 instructions per entry
-    findIndex<T>(list, func(x) = equal(element, x))
+    let blocks = list.blocks;
```

Comment:
Make code comments saying that code is copied from findIndex. Then the reviewer only has to compare code, not think through it twice.